examples/platforms/posix/random.c

```diff
@@ -75,6 +88,40 @@ ThreadError otPlatRandomSecureGet(uint16_t aInputLength, uint8_t *aOutput, uint1
 {
     ThreadError error = kThreadError_None;
 
+#if __SANITIZE_ADDRESS__ == 0
+
+    FILE *file = NULL;
+    size_t readLength;
+
+    *aOutputLength = 0;
+
+    otEXPECT_ACTION(aOutput && aOutputLength, error = kThreadError_InvalidArgs);
+
+    file = fopen("/dev/urandom", "rb");
+    otEXPECT_ACTION(file != NULL, error = kThreadError_Failed);
+
+    readLength = fread(aOutput, 1, aInputLength, file);
+    otEXPECT_ACTION(readLength == aInputLength, error = kThreadError_Failed);
```

Comment:
Good point.  The declaration was modeled off of `mbedtls_hardware_poll()`.  However, mbedtls also errors when the desired length cannot be generated.  I'm thinking of updating the documentation (and possibly this API) to simply return error if the requested length cannot be provided.  Thoughts?